### PR TITLE
[Merged by Bors] - Tortoise code cleanup

### PR DESCRIPTION
**Pull request by almogdepaz**
## Motivation
code cleanup

## Changes
cosmetic

## Test Plan
same

### CODE REVIEW

File: tortoise/ninja_tortoise.go [hunk 1]
```diff
@@ -286,53 +282,53 @@ func getId(bids []types.BlockID) PatternId {
...
+	ni.db.ForBlockInView(tp, bottomOfWindow, foo)
 }
 
-func (ni *NinjaTortoise) updatePatternTally(newMinGood votingPattern, correctionMap map[types.BlockID]vec, effCountMap map[types.LayerID]int) {
-	ni.Debug("update tally pbase id:%s layer:%s p id:%s layer:%s", ni.PBase.id, ni.PBase.Layer(), newMinGood.id, newMinGood.Layer())
+func (ni *ninjaTortoise) updatePatternTally(newMinGood votingPattern, correctionMap map[types.BlockID]vec, effCountMap map[types.LayerID]int) {
+	ni.logger.Debug("update tally pbase id:%s layer:%s p id:%s layer:%s", ni.PBase.id, ni.PBase.Layer(), newMinGood.id, newMinGood.Layer())
 	for idx, effc := range effCountMap {
 		g := ni.TGood[idx]
 		for b, v := range ni.TVote[g] {
```
Comment (barakshani) [inline tortoise/ninja_tortoise.go:334]: please change comment below to:
// g's opinion -> p's tally

File: tortoise/ninja_tortoise.go [hunk 2]
```diff
@@ -343,7 +339,7 @@ func (ni *NinjaTortoise) updatePatternTally(newMinGood votingPattern, correction
 	}
 
 	for b, tally := range ni.TTally[newMinGood] {
-		if count, found := correctionMap[b.Id()]; found {
+		if count, found := correctionMap[b.id()]; found {
 			ni.TTally[newMinGood][b] = tally.Add(count)
 		}
 	}
```
Comment (barakshani) [inline tortoise/ninja_tortoise.go:346]: Please add a comment about this logic

File: tortoise/ninja_tortoise.go [hunk 3]
```diff
@@ -555,7 +552,7 @@ func (ni *NinjaTortoise) handleIncomingLayer(newlyr *types.Layer) {
 			}
 
 			tp := ni.TPattern[p]
-			ni.ForBlockInView(tp, windowStart, foo)
+			ni.db.ForBlockInView(tp, windowStart, foo)
 
 			//add corrected implicit votes
```
Comment (barakshani) [inline tortoise/ninja_tortoise.go:558]: why "corrected"?

Comment (almogdepaz) [reply tortoise/ninja_tortoise.go:558]: because we take into account the correction vectors we prepared we are not just adding implicit.

Comment (barakshani) [reply tortoise/ninja_tortoise.go:558]: I see, `updateCorrectionVectors()` only "prepares the ground" for the corrections, but here (in lines 341 to 345) we actually apply them?

Comment (almogdepaz) [reply tortoise/ninja_tortoise.go:558]: yes

Comment (barakshani) [reply tortoise/ninja_tortoise.go:558]: feel like commenting about it somewhere?

File: tortoise/algorithm.go [hunk 4]
```diff
@@ -49,24 +52,32 @@ func (alg *Algorithm) HandleLateBlock(b *types.Block) (types.LayerID, types.Laye
 	return oldPbase, newPbase
 }
 
-func (alg *Algorithm) PersistTortoise() error {
-	alg.Lock()
-	defer alg.Unlock()
+//Persist saves a copy of the current tortoise state to the database
+func (alg *tortoise) Persist() error {
```
Comment (gavraz) [inline tortoise/algorithm.go]: maybe rename alg to trtl or something like that?

File: tortoise/ninja_tortoise.go [hunk 5]
```diff
@@ -180,51 +176,51 @@ func (ni *NinjaTortoise) evictOutOfPbase() {
...
 			delete(ni.Patterns, lyr)
 		}()
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			ids, err := ni.LayerBlockIds(lyr)
+			ids, err := ni.db.LayerBlockIds(lyr)
 			if err != nil {
-				ni.Error("could not get layer ids for layer %v %v", lyr, err)
+				ni.logger.Error("could not get layer ids for layer %v %v", lyr, err)
```
Comment (gavraz) [inline tortoise/ninja_tortoise.go]: add "with" to your logs and use the JSON format variable

**Review by gavraz — APPROVED** [review]
See my comments

**Comment by almogdepaz** [comment]
bors merge

**Comment by almogdepaz** [comment]
bors -r

**Comment by almogdepaz** [comment]
bors r_

**Comment by almogdepaz** [comment]
bors merge

File: tortoise/algorithm.go (hunk 4 above)
Comment (almogdepaz) [reply tortoise/algorithm.go]: good idea

**Pull request closed without merging**